2fa (#135)
pretix/pretix-docs: docs/guides/two-factor-authentication.md

# Two-factor authentication 

By default, your pretix account is secured with a single factor: your password. 
You can add a second factor to improve security. 
This is called "two-factor authentication" or **2FA**. 

The second security factor can be a WebAuthn-compatible hardware token or an authenticator application. 
This article explains the advantages of 2FA and how to set it up for your pretix user account. 

## Background information 

Depending on your usage of pretix and permission settings, your user account can grant access to a lot of sensitive information. 

> This sensitive information can include, for example, company secrets, payment history, and your customers' contact data. 

Reviewer: > This sensitive information can include, for example, company secrets, payment history, and your customers' contact data. 

Is 'contact' even necessary here? I would drop it as one should be afraid to loose any and all data of customers.

A strong password offers a good baseline level of security. 
But potential malicious actors can breach passwords through malware, social engineering, and other methods. 

A second factor for authentication makes such an attack significantly more difficult. 
An attacker would have to gain access to both your 2FA device and your account credentials. 
Detecting a breach of one security factor can give you enough time to change it before the attacker can also breach the other factor. 

We recommend setting up 2FA for all user accounts at your organization. 
If you want to require your coworkers to use 2FA for their pretix accounts, refer to the article on [Teams](teams.md#requiring-two-factor-authentication-2fa). 

> Once everyone on your team has done so, a malicious attack is very unlikely to succeed. 

Author: Vorschlag: 
```suggestion
We recommend setting up 2FA for all user accounts within your organization. 
The IT security standard ISO27001 requires secure authentication and recommends 2FA. 
If you want to require your coworkers to use 2FA for their pretix accounts, refer to the article on [Teams](teams.md#requiring-two-factor-authentication-2fa). 
Once everyone on your team has done so, a malicious attack is very unlikely to succeed. 
```

Die Alternative wäre 

```suggestion
The IT security standard ISO27001 requires secure authentication and recommends multi-factor authentication. 
2FA is a form of multi-factor authentication. 
```
aber das ist mir zu sperrig, daher lieber die verkürzte Variante

## Prerequisites

Depending on the authentication method you intend to use, you need to have: 

>  - a WebAuthn-compatible hardware token such as a Yubikey **or** 

Author: @leiascyr do you have an opinion on this? The backend uses "Yubikey" everywhere AFAICS. The word only appears on this page of the docs so far.

>  - an authenticator app capable of generating time-based one-time passwords (TOTP), such as Microsoft Authenticator, Google Authenticator, or Bitwarden. 

Author: Dann brauchen wir hier zumindest einen Hinweis: 

```suggestion

## Prerequisites

!!! Note 
    The applications and devices listed here are **examples**. 
    They are not recommendations. 
    We recommend doing some research into trustworthy hardware tokens or authenticator apps. 
    Do this before deciding on a 2FA solution for your team. 

Depending on the authentication method you intend to use, you need to have: 

 - a WebAuthn-compatible hardware token such as a Yubikey **or** 
 - an authenticator app capable of generating time-based one-time passwords (TOTP), such as Microsoft Authenticator, Google Authenticator, Bitwarden, or FreeOTP. 
```

    However, every additional method for logging in to your account offers potential attackers an additional point of entry. 
    In order to maximize security, only add a single device for 2FA. 

The advantage of a TOTP app is that you probably already have a device on hand which you can use for TOTP. 
This can be your work phone, your personal phone, another mobile device, or a separate computer. 
If this is the case, then you can use the device you already have on hand for 2FA. 

If you use separate applications for password storage and TOTP generation, then the TOTP method offers good security. 
However, like all software run on a device connected to the internet, a TOTP app is vulnerable to malware. 

A hardware token is much cheaper than a smartphone. 
If you need to buy a new device for 2FA, then this can be an advantage. 
Another advantage is that hardware tokens are less susceptible to software attacks. 

You must remove your hardware token from the computer and keep it on your person whenever you are not actively using it. 
One downside of the hardware token is that it is typically smaller than a phone and only used for authentication purposes. 

> Thus, it can potentially be easier to lose. 

Reviewer: > Thus, it can potentially be easier to lose. 

Thus, they are easier to lose.

# Two-factor authentication 

> The term "Two-factor authentication" or **2FA** refers to using two factors to authenticate your identity. 

Reviewer 2: > The term "Two-factor authentication" or **2FA** refers to using two factors to authenticate your identity. 

Das liest sich sehr redundant, bekommen wir das informativer formuliert?

This provides more security than authenticating with a single factor such as a password. 

You can secure your pretix user account with 2FA. 
The first factor is always your password. 
The second factor can be a WebAuthn-compatible hardware token or an authenticator application. 

This article explains the advantages of 2FA and how to set it up for your pretix user account. 

## Background information 

Depending on your usage of pretix and permission settings, your user account can grant access to a lot of sensitive information. 

> This sensitive information can include, for example, company secrets, payment history, and your customer's contact data. 

Reviewer 2: ```suggestion
This sensitive information can include, for example, company secrets, payment history, and your customers' contact data. 
```

Mutmaßlich gibt es ja mehr als 1 Endkunden

A strong password offers a good baseline level of security. 
But potential malicious actors can breach passwords through malware, social engineering, and other methods. 

A second factor for authentication makes such an attack significantly more difficult. 
An attacker would have to gain access to both your 2FA device and your account credentials. 
Detecting a breach of one security factor can give you enough time to change it before the attacker can also breach the other factor. 

> We recommend setting up 2FA for all user account at your organization. 

Reviewer 2: ```suggestion
We recommend setting up 2FA for all user accounts at your organization. 
```

If you want to require your coworkers to use 2FA for their pretix accounts, refer to the article on [Teams](teams.md#requiring-two-factor-authentication-2fa). 
Once everyone on your team has done so, a malicious attack is almost impossible to succeed. 

## Prerequisites

Depending on the authentication method you intend to use, you need to have: 

 - a WebAuthn-compatible hardware token such as a Yubikey **or** 
 - an authenticator app capable of generating time-based one-time passwords (TOTP), such as Microsoft Authenticator, Google Authenticator, or Bitwarden. 

## How to

Log in to your pretix user account. 

> Click the :btn-icon:fa3-user: [Your name]: button in the top right corner. 

Reviewer 2: ```suggestion
Click the :btn-icon:fa3-user:[Your name]: button in the top right corner. 
```

!["Page titled 'Account settings', displaying settings for 'Full name', 'Language', 'Default timezone', 'Notifications', 'Email', 'Password', 'Two-factor authentication', 'Authorized applications', and 'Account history'."](../assets/screens/account/account-settings.png "Account settings")

Before you set up 2FA, you should store your **emergency tokens** outside of pretix. 
You can use these codes in place of your 2FA device in case you lose access to it. 
Copy the codes and print them or write them down on paper. 
Store that paper in a safe place such as a locked drawer. 

Alternatively, you can store the emergency tokens digitally in an encrypted file or in a password manager. 
**Do not** store them in the same password manager or database that also contains your account password. 

!!! Note 
    Ensure that you have stored your emergency tokens outside of pretix and that you know where to find them. 
    You need these tokens in case you lose access to your 2FA device. 

Take a look at the "Two-factor authentication" setting. 
If 2FA is disabled, then there is a gray "Disabled" tag next to it. 
In order to enable 2FA, click the :btn:Enable: button. 

> !["Page titled 'Two-factor authentication', displaying the 'Two factor status' as 'currently' disabled, a button for adding a new registered device, and emergency tokens."](../assets/screens/account/two-factor-authentication.png "Two-factor authentication")

Reviewer 2: ```suggestion
!["Page titled 'Two-factor authentication', displaying the 'Two factor status' as 'currently disabled', a button for adding a new registered device, and emergency tokens."](../assets/screens/account/two-factor-authentication.png "Two-factor authentication")
```

> On the next page, click the :btn-icon:fa3-plus: Add a new device: button. 

Reviewer 2: ```suggestion
On the next page, click the :btn-icon:fa3-plus:Add a new device: button. 
```
